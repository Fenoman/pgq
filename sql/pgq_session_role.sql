\set VERBOSITY 'terse'
set client_min_messages = 'warning';

show session_replication_role;

select pgq.create_queue('role_test_enabled');
select pgq.create_queue('role_test_disabled');
update pgq.queue set queue_disable_insert=true where queue_name = 'role_test_disabled';

select pgq.insert_event('role_test_enabled', 'enabled', 'role:origin');
select pgq.insert_event('role_test_disabled', 'disabled', 'role:origin');

set session_replication_role = 'replica';
show session_replication_role;
select pgq.insert_event('role_test_enabled', 'enabled', 'role:replica');
select pgq.insert_event('role_test_disabled', 'disabled', 'role:replica');

set session_replication_role = 'local';
show session_replication_role;
select pgq.insert_event('role_test_enabled', 'enabled', 'role:local');
select pgq.insert_event('role_test_disabled', 'disabled', 'role:local');

set session_replication_role = 'origin';
show session_replication_role;
select pgq.insert_event('role_test_enabled', 'enabled', 'role:origin');
select pgq.insert_event('role_test_disabled', 'disabled', 'role:origin');

select pgq.drop_queue('role_test_enabled');
select pgq.drop_queue('role_test_disabled');

