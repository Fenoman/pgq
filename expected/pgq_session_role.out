\set VERBOSITY 'terse'
set client_min_messages = 'warning';
show session_replication_role;
 session_replication_role 
--------------------------
 origin
(1 row)

select pgq.create_queue('role_test_enabled');
 create_queue 
--------------
            1
(1 row)

select pgq.create_queue('role_test_disabled');
 create_queue 
--------------
            1
(1 row)

update pgq.queue set queue_disable_insert=true where queue_name = 'role_test_disabled';
select pgq.insert_event('role_test_enabled', 'enabled', 'role:origin');
 insert_event 
--------------
            1
(1 row)

select pgq.insert_event('role_test_disabled', 'disabled', 'role:origin');
ERROR:  Insert into queue disallowed
set session_replication_role = 'replica';
show session_replication_role;
 session_replication_role 
--------------------------
 replica
(1 row)

select pgq.insert_event('role_test_enabled', 'enabled', 'role:replica');
 insert_event 
--------------
            2
(1 row)

select pgq.insert_event('role_test_disabled', 'disabled', 'role:replica');
 insert_event 
--------------
            2
(1 row)

set session_replication_role = 'local';
show session_replication_role;
 session_replication_role 
--------------------------
 local
(1 row)

select pgq.insert_event('role_test_enabled', 'enabled', 'role:local');
 insert_event 
--------------
            3
(1 row)

select pgq.insert_event('role_test_disabled', 'disabled', 'role:local');
ERROR:  Insert into queue disallowed
set session_replication_role = 'origin';
show session_replication_role;
 session_replication_role 
--------------------------
 origin
(1 row)

select pgq.insert_event('role_test_enabled', 'enabled', 'role:origin');
 insert_event 
--------------
            4
(1 row)

select pgq.insert_event('role_test_disabled', 'disabled', 'role:origin');
ERROR:  Insert into queue disallowed
select pgq.drop_queue('role_test_enabled');
 drop_queue 
------------
          1
(1 row)

select pgq.drop_queue('role_test_disabled');
 drop_queue 
------------
          1
(1 row)

